<?xml version="1.0" encoding="UTF-8" ?>
<launch>
    <!-- downsample messages of a rostopic and store them to a new rosbag -->
    <node pkg="ikalibr" type="ikalibr_bag_topic_downsample" name="ikalibr_bag_topic_downsample" output="screen">
        <!-- the input rosbag -->
        <param name="input_bag_path" value="/home/csl/dataset/vector/desk_fast/desk_fast1.synced.left_camera.bag"
               type="string"/>
        <!-- the rostopic to down sampled -->
        <param name="topic_to_downsample" value="/camera/left/image_mono" type="string"/>
        <!-- the desired ros topic frequency -->
        <param name="desired_frequency" value="10" type="double"/>
        <!-- the output rosbag -->
        <param name="output_bag_path" value="/home/csl/dataset/vector/desk_fast/desk_fast1.synced.left_camera_10hz.bag"
               type="string"/>
    </node>

    <!--
         iKalibr: Unified Targetless Spatiotemporal Calibration Framework
         Copyright 2024, the School of Geodesy and Geomatics (SGG), Wuhan University, China
         https://github.com/Unsigned-Long/iKalibr.git

         Author: Shuolong Chen (shlchen@whu.edu.cn)
         GitHub: https://github.com/Unsigned-Long
          ORCID: 0000-0002-5283-9057

         Purpose: See .h/.hpp file.

         Redistribution and use in source and binary forms, with or without
         modification, are permitted provided that the following conditions are met:

         * Redistributions of source code must retain the above copyright notice,
           this list of conditions and the following disclaimer.
         * Redistributions in binary form must reproduce the above copyright notice,
           this list of conditions and the following disclaimer in the documentation
           and/or other materials provided with the distribution.
         * The names of its contributors can not be
           used to endorse or promote products derived from this software without
           specific prior written permission.

         THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
         AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
         IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
         ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
         LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
         CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
         SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
         INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
         CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
         ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
         POSSIBILITY OF SUCH DAMAGE.
    -->
</launch>